File: Striver A2Z/Step 4 - Binary Search [1D, 2D arrays, Search Space]/Lec 1 - BS on 1D Arrays/search-insert-position.cpp

// https://leetcode.com/problems/search-insert-position/

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();
        int insertIndex = n;
        if(n == 0) return insertIndex;

        int start = 0, end = n-1;
        while(start <= end){
            int mid = (start + end)>>1;
            if(nums[mid] >= target){
                insertIndex = mid;
                end = mid-1;
            }else start = mid+1;
        }
        return (nums[0] >= target) ? 0 : insertIndex;
    }
};

File: Striver A2Z/Step 4 - Binary Search [1D, 2D arrays, Search Space]/Lec 1 - BS on 1D Arrays/search-insert-position.js

// https://leetcode.com/problems/search-insert-position/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let len = nums.length;
    let insertIndex = len;

    let start = 0, end = len-1;
    while(start <= end){
        let mid = Math.floor((start + end)/2);
        if(nums[mid] >= target){
            insertIndex = mid;
            end = mid-1;
        }else start = mid+1;
    }

    return insertIndex;
};